// ui-verse/index.mjs
import { jsx as g, jsxs as F, Fragment as Vt } from "react/jsx-runtime";
import Qr, { createContext as Xt, useContext as ht, useState as Y, useRef as at, useEffect as B, useLayoutEffect as Kr, useMemo as Zr, useCallback as bt, useId as Jr, forwardRef as it, Fragment as te, createElement as re } from "react";
import ee from "react-dom";
const oo = (t) => /* @__PURE__ */ g("h1", { ...t, children: t.children });
function P(t) {
  return Object.keys(t);
}
function Pt(t) {
  return t && typeof t == "object" && !Array.isArray(t);
}
function Ut(t, r) {
  const e = { ...t }, n = r;
  return Pt(t) && Pt(r) && Object.keys(r).forEach((a) => {
    Pt(n[a]) && a in t ? e[a] = Ut(e[a], n[a]) : e[a] = n[a];
  }), e;
}
function ne(t) {
  return t.replace(/[A-Z]/g, (r) => `-${r.toLowerCase()}`);
}
function oe(t) {
  var r;
  return typeof t != "string" || !t.includes("var(--mantine-scale)") ? t : (r = t.match(/^calc\((.*?)\)$/)) == null ? void 0 : r[1].split("*")[0].trim();
}
function ae(t) {
  const r = oe(t);
  return typeof r == "number" ? r : typeof r == "string" ? r.includes("calc") || r.includes("var") ? r : r.includes("px") ? Number(r.replace("px", "")) : r.includes("rem") ? Number(r.replace("rem", "")) * 16 : r.includes("em") ? Number(r.replace("em", "")) * 16 : Number(r) : NaN;
}
function Lt(t) {
  return t === "0rem" ? "0rem" : `calc(${t} * var(--mantine-scale))`;
}
function _r(t, { shouldScale: r = !1 } = {}) {
  function e(n) {
    if (n === 0 || n === "0")
      return `0${t}`;
    if (typeof n == "number") {
      const a = `${n / 16}${t}`;
      return r ? Lt(a) : a;
    }
    if (typeof n == "string") {
      if (n === "" || n.startsWith("calc(") || n.startsWith("clamp(") || n.includes("rgba("))
        return n;
      if (n.includes(","))
        return n.split(",").map((o) => e(o)).join(",");
      if (n.includes(" "))
        return n.split(" ").map((o) => e(o)).join(" ");
      if (n.includes(t))
        return r ? Lt(n) : n;
      const a = n.replace("px", "");
      if (!Number.isNaN(Number(a))) {
        const o = `${Number(a) / 16}${t}`;
        return r ? Lt(o) : o;
      }
    }
    return n;
  }
  return e;
}
const c = _r("rem", { shouldScale: !0 }), lr = _r("em");
function qt(t) {
  return Object.keys(t).reduce((r, e) => (t[e] !== void 0 && (r[e] = t[e]), r), {});
}
function Nr(t) {
  return typeof t == "number" ? !0 : typeof t == "string" ? t.startsWith("calc(") || t.startsWith("var(") || t.includes(" ") && t.trim() !== "" ? !0 : /[0-9]/.test(t.trim().replace("-", "")[0]) : !1;
}
function ie(t = null) {
  const r = Xt(t);
  return [({ children: a, value: o }) => /* @__PURE__ */ g(r.Provider, { value: o, children: a }), () => ht(r)];
}
function G(t, r = "size", e = !0) {
  if (t !== void 0)
    return Nr(t) ? e ? c(t) : t : `var(--${r}-${t})`;
}
function Qt(t) {
  return t === void 0 ? "var(--mantine-radius-default)" : G(t, "mantine-radius");
}
function W(t) {
  return G(t, "mantine-font-size");
}
function se() {
  return `mantine-${Math.random().toString(36).slice(2, 11)}`;
}
function ce(t, r) {
  try {
    return t.addEventListener("change", r), () => t.removeEventListener("change", r);
  } catch {
    return t.addListener(r), () => t.removeListener(r);
  }
}
function le(t, r) {
  return typeof window < "u" && "matchMedia" in window ? window.matchMedia(t).matches : !1;
}
function de(t, r, { getInitialValueInEffect: e } = {
  getInitialValueInEffect: !0
}) {
  const [n, a] = Y(
    e ? r : le(t)
  ), o = at();
  return B(() => {
    if ("matchMedia" in window)
      return o.current = window.matchMedia(t), a(o.current.matches), ce(o.current, (i) => a(i.matches));
  }, [t]), n;
}
const Kt = typeof document < "u" ? Kr : B;
function ue(t, r) {
  const e = at(!1);
  B(
    () => () => {
      e.current = !1;
    },
    []
  ), B(() => {
    if (e.current)
      return t();
    e.current = !0;
  }, r);
}
const fe = Qr.useId || (() => {
});
function pe() {
  const t = fe();
  return t ? `mantine-${t.replace(/:/g, "")}` : "";
}
function me(t) {
  const r = pe(), [e, n] = Y(r);
  return Kt(() => {
    n(se());
  }, []), typeof t == "string" ? t : typeof window > "u" ? r : e;
}
function ye(t, r) {
  return de("(prefers-reduced-motion: reduce)", t, r);
}
function kr(t) {
  var r, e, n = "";
  if (typeof t == "string" || typeof t == "number")
    n += t;
  else if (typeof t == "object")
    if (Array.isArray(t)) {
      var a = t.length;
      for (r = 0; r < a; r++)
        t[r] && (e = kr(t[r])) && (n && (n += " "), n += e);
    } else
      for (e in t)
        t[e] && (n && (n += " "), n += e);
  return n;
}
function X() {
  for (var t, r, e = 0, n = "", a = arguments.length; e < a; e++)
    (t = arguments[e]) && (r = kr(t)) && (n && (n += " "), n += r);
  return n;
}
const ge = {};
function be(t) {
  const r = {};
  return t.forEach((e) => {
    Object.entries(e).forEach(([n, a]) => {
      r[n] ? r[n] = X(r[n], a) : r[n] = a;
    });
  }), r;
}
function Zt({ theme: t, classNames: r, props: e, stylesCtx: n }) {
  const o = (Array.isArray(r) ? r : [r]).map(
    (i) => typeof i == "function" ? i(t, e, n) : i || ge
  );
  return be(o);
}
function Ht({ theme: t, styles: r, props: e, stylesCtx: n }) {
  return (Array.isArray(r) ? r : [r]).reduce((o, i) => typeof i == "function" ? { ...o, ...i(t, e, n) } : { ...o, ...i }, {});
}
const Ir = Xt(null);
function U() {
  const t = ht(Ir);
  if (!t)
    throw new Error("[@mantine/core] MantineProvider was not found in tree");
  return t;
}
function he() {
  return U().cssVariablesResolver;
}
function ve() {
  return U().classNamesPrefix;
}
function Jt() {
  return U().getStyleNonce;
}
function $e() {
  return U().withStaticClasses;
}
function Se() {
  return U().headless;
}
function we() {
  var t;
  return (t = U().stylesTransform) == null ? void 0 : t.sx;
}
function Ce() {
  var t;
  return (t = U().stylesTransform) == null ? void 0 : t.styles;
}
function _e(t) {
  return /^#?([0-9A-F]{3}){1,2}([0-9A-F]{2})?$/i.test(t);
}
function Ne(t) {
  let r = t.replace("#", "");
  if (r.length === 3) {
    const i = r.split("");
    r = [
      i[0],
      i[0],
      i[1],
      i[1],
      i[2],
      i[2]
    ].join("");
  }
  if (r.length === 8) {
    const i = parseInt(r.slice(6, 8), 16) / 255;
    return {
      r: parseInt(r.slice(0, 2), 16),
      g: parseInt(r.slice(2, 4), 16),
      b: parseInt(r.slice(4, 6), 16),
      a: i
    };
  }
  const e = parseInt(r, 16), n = e >> 16 & 255, a = e >> 8 & 255, o = e & 255;
  return {
    r: n,
    g: a,
    b: o,
    a: 1
  };
}
function ke(t) {
  const [r, e, n, a] = t.replace(/[^0-9,./]/g, "").split(/[/,]/).map(Number);
  return { r, g: e, b: n, a: a || 1 };
}
function Ie(t) {
  const r = /^hsla?\(\s*(\d+)\s*,\s*(\d+%)\s*,\s*(\d+%)\s*(,\s*(0?\.\d+|\d+(\.\d+)?))?\s*\)$/i, e = t.match(r);
  if (!e)
    return {
      r: 0,
      g: 0,
      b: 0,
      a: 1
    };
  const n = parseInt(e[1], 10), a = parseInt(e[2], 10) / 100, o = parseInt(e[3], 10) / 100, i = e[5] ? parseFloat(e[5]) : void 0, s = (1 - Math.abs(2 * o - 1)) * a, l = n / 60, d = s * (1 - Math.abs(l % 2 - 1)), u = o - s / 2;
  let f, p, y;
  return l >= 0 && l < 1 ? (f = s, p = d, y = 0) : l >= 1 && l < 2 ? (f = d, p = s, y = 0) : l >= 2 && l < 3 ? (f = 0, p = s, y = d) : l >= 3 && l < 4 ? (f = 0, p = d, y = s) : l >= 4 && l < 5 ? (f = d, p = 0, y = s) : (f = s, p = 0, y = d), {
    r: Math.round((f + u) * 255),
    g: Math.round((p + u) * 255),
    b: Math.round((y + u) * 255),
    a: i || 1
  };
}
function tr(t) {
  return _e(t) ? Ne(t) : t.startsWith("rgb") ? ke(t) : t.startsWith("hsl") ? Ie(t) : {
    r: 0,
    g: 0,
    b: 0,
    a: 1
  };
}
function mt(t, r) {
  if (t.startsWith("var("))
    return `color-mix(in srgb, ${t}, black ${r * 100}%)`;
  const { r: e, g: n, b: a, a: o } = tr(t), i = 1 - r, s = (l) => Math.round(l * i);
  return `rgba(${s(e)}, ${s(n)}, ${s(a)}, ${o})`;
}
function ut(t, r) {
  return typeof t.primaryShade == "number" ? t.primaryShade : r === "dark" ? t.primaryShade.dark : t.primaryShade.light;
}
function At(t) {
  return t <= 0.03928 ? t / 12.92 : ((t + 0.055) / 1.055) ** 2.4;
}
function xe(t) {
  const r = t.match(/oklch\((.*?)%\s/);
  return r ? parseFloat(r[1]) : null;
}
function Re(t) {
  if (t.startsWith("oklch("))
    return (xe(t) || 0) / 100;
  const { r, g: e, b: n } = tr(t), a = r / 255, o = e / 255, i = n / 255, s = At(a), l = At(o), d = At(i);
  return 0.2126 * s + 0.7152 * l + 0.0722 * d;
}
function lt(t, r = 0.179) {
  return t.startsWith("var(") ? !1 : Re(t) > r;
}
function ft({
  color: t,
  theme: r,
  colorScheme: e
}) {
  if (typeof t != "string")
    throw new Error(
      `[@mantine/core] Failed to parse color. Expected color to be a string, instead got ${typeof t}`
    );
  if (t === "bright")
    return {
      color: t,
      value: e === "dark" ? r.white : r.black,
      shade: void 0,
      isThemeColor: !1,
      isLight: lt(
        e === "dark" ? r.white : r.black,
        r.luminanceThreshold
      ),
      variable: "--mantine-color-bright"
    };
  if (t === "dimmed")
    return {
      color: t,
      value: e === "dark" ? r.colors.dark[2] : r.colors.gray[7],
      shade: void 0,
      isThemeColor: !1,
      isLight: lt(
        e === "dark" ? r.colors.dark[2] : r.colors.gray[6],
        r.luminanceThreshold
      ),
      variable: "--mantine-color-dimmed"
    };
  if (t === "white" || t === "black")
    return {
      color: t,
      value: t === "white" ? r.white : r.black,
      shade: void 0,
      isThemeColor: !1,
      isLight: lt(
        t === "white" ? r.white : r.black,
        r.luminanceThreshold
      ),
      variable: `--mantine-color-${t}`
    };
  const [n, a] = t.split("."), o = a ? Number(a) : void 0, i = n in r.colors;
  if (i) {
    const s = o !== void 0 ? r.colors[n][o] : r.colors[n][ut(r, e || "light")];
    return {
      color: n,
      value: s,
      shade: o,
      isThemeColor: i,
      isLight: lt(s, r.luminanceThreshold),
      variable: a ? `--mantine-color-${n}-${o}` : `--mantine-color-${n}-filled`
    };
  }
  return {
    color: t,
    value: t,
    isThemeColor: i,
    isLight: lt(t, r.luminanceThreshold),
    shade: o,
    variable: void 0
  };
}
function Dt(t, r) {
  const e = ft({ color: t || r.primaryColor, theme: r });
  return e.variable ? `var(${e.variable})` : t;
}
function dr(t, r) {
  const e = {
    from: (t == null ? void 0 : t.from) || r.defaultGradient.from,
    to: (t == null ? void 0 : t.to) || r.defaultGradient.to,
    deg: (t == null ? void 0 : t.deg) || r.defaultGradient.deg || 0
  }, n = Dt(e.from, r), a = Dt(e.to, r);
  return `linear-gradient(${e.deg}deg, ${n} 0%, ${a} 100%)`;
}
function z(t, r) {
  if (typeof t != "string" || r > 1 || r < 0)
    return "rgba(0, 0, 0, 1)";
  if (t.startsWith("var(")) {
    const o = (1 - r) * 100;
    return `color-mix(in srgb, ${t}, transparent ${o}%)`;
  }
  if (t.startsWith("oklch"))
    return t.includes("/") ? t.replace(/\/\s*[\d.]+\s*\)/, `/ ${r})`) : t.replace(")", ` / ${r})`);
  const { r: e, g: n, b: a } = tr(t);
  return `rgba(${e}, ${n}, ${a}, ${r})`;
}
const et = z, Te = ({
  color: t,
  theme: r,
  variant: e,
  gradient: n,
  autoContrast: a
}) => {
  const o = ft({ color: t, theme: r }), i = typeof a == "boolean" ? a : r.autoContrast;
  if (e === "filled") {
    const s = i && o.isLight ? "var(--mantine-color-black)" : "var(--mantine-color-white)";
    return o.isThemeColor ? o.shade === void 0 ? {
      background: `var(--mantine-color-${t}-filled)`,
      hover: `var(--mantine-color-${t}-filled-hover)`,
      color: s,
      border: `${c(1)} solid transparent`
    } : {
      background: `var(--mantine-color-${o.color}-${o.shade})`,
      hover: `var(--mantine-color-${o.color}-${o.shade === 9 ? 8 : o.shade + 1})`,
      color: s,
      border: `${c(1)} solid transparent`
    } : {
      background: t,
      hover: mt(t, 0.1),
      color: s,
      border: `${c(1)} solid transparent`
    };
  }
  if (e === "light") {
    if (o.isThemeColor) {
      if (o.shade === void 0)
        return {
          background: `var(--mantine-color-${t}-light)`,
          hover: `var(--mantine-color-${t}-light-hover)`,
          color: `var(--mantine-color-${t}-light-color)`,
          border: `${c(1)} solid transparent`
        };
      const s = r.colors[o.color][o.shade];
      return {
        background: z(s, 0.1),
        hover: z(s, 0.12),
        color: `var(--mantine-color-${o.color}-${Math.min(o.shade, 6)})`,
        border: `${c(1)} solid transparent`
      };
    }
    return {
      background: z(t, 0.1),
      hover: z(t, 0.12),
      color: t,
      border: `${c(1)} solid transparent`
    };
  }
  if (e === "outline")
    return o.isThemeColor ? o.shade === void 0 ? {
      background: "transparent",
      hover: `var(--mantine-color-${t}-outline-hover)`,
      color: `var(--mantine-color-${t}-outline)`,
      border: `${c(1)} solid var(--mantine-color-${t}-outline)`
    } : {
      background: "transparent",
      hover: z(r.colors[o.color][o.shade], 0.05),
      color: `var(--mantine-color-${o.color}-${o.shade})`,
      border: `${c(1)} solid var(--mantine-color-${o.color}-${o.shade})`
    } : {
      background: "transparent",
      hover: z(t, 0.05),
      color: t,
      border: `${c(1)} solid ${t}`
    };
  if (e === "subtle") {
    if (o.isThemeColor) {
      if (o.shade === void 0)
        return {
          background: "transparent",
          hover: `var(--mantine-color-${t}-light-hover)`,
          color: `var(--mantine-color-${t}-light-color)`,
          border: `${c(1)} solid transparent`
        };
      const s = r.colors[o.color][o.shade];
      return {
        background: "transparent",
        hover: z(s, 0.12),
        color: `var(--mantine-color-${o.color}-${Math.min(o.shade, 6)})`,
        border: `${c(1)} solid transparent`
      };
    }
    return {
      background: "transparent",
      hover: z(t, 0.12),
      color: t,
      border: `${c(1)} solid transparent`
    };
  }
  return e === "transparent" ? o.isThemeColor ? o.shade === void 0 ? {
    background: "transparent",
    hover: "transparent",
    color: `var(--mantine-color-${t}-light-color)`,
    border: `${c(1)} solid transparent`
  } : {
    background: "transparent",
    hover: "transparent",
    color: `var(--mantine-color-${o.color}-${Math.min(o.shade, 6)})`,
    border: `${c(1)} solid transparent`
  } : {
    background: "transparent",
    hover: "transparent",
    color: t,
    border: `${c(1)} solid transparent`
  } : e === "white" ? o.isThemeColor ? o.shade === void 0 ? {
    background: "var(--mantine-color-white)",
    hover: mt(r.white, 0.01),
    color: `var(--mantine-color-${t}-filled)`,
    border: `${c(1)} solid transparent`
  } : {
    background: "var(--mantine-color-white)",
    hover: mt(r.white, 0.01),
    color: `var(--mantine-color-${o.color}-${o.shade})`,
    border: `${c(1)} solid transparent`
  } : {
    background: "var(--mantine-color-white)",
    hover: mt(r.white, 0.01),
    color: t,
    border: `${c(1)} solid transparent`
  } : e === "gradient" ? {
    background: dr(n, r),
    hover: dr(n, r),
    color: "var(--mantine-color-white)",
    border: "none"
  } : e === "default" ? {
    background: "var(--mantine-color-default)",
    hover: "var(--mantine-color-default-hover)",
    color: "var(--mantine-color-default-color)",
    border: `${c(1)} solid var(--mantine-color-default-border)`
  } : {};
}, Me = {
  dark: [
    "#C9C9C9",
    "#b8b8b8",
    "#828282",
    "#696969",
    "#424242",
    "#3b3b3b",
    "#2e2e2e",
    "#242424",
    "#1f1f1f",
    "#141414"
  ],
  gray: [
    "#f8f9fa",
    "#f1f3f5",
    "#e9ecef",
    "#dee2e6",
    "#ced4da",
    "#adb5bd",
    "#868e96",
    "#495057",
    "#343a40",
    "#212529"
  ],
  red: [
    "#fff5f5",
    "#ffe3e3",
    "#ffc9c9",
    "#ffa8a8",
    "#ff8787",
    "#ff6b6b",
    "#fa5252",
    "#f03e3e",
    "#e03131",
    "#c92a2a"
  ],
  pink: [
    "#fff0f6",
    "#ffdeeb",
    "#fcc2d7",
    "#faa2c1",
    "#f783ac",
    "#f06595",
    "#e64980",
    "#d6336c",
    "#c2255c",
    "#a61e4d"
  ],
  grape: [
    "#f8f0fc",
    "#f3d9fa",
    "#eebefa",
    "#e599f7",
    "#da77f2",
    "#cc5de8",
    "#be4bdb",
    "#ae3ec9",
    "#9c36b5",
    "#862e9c"
  ],
  violet: [
    "#f3f0ff",
    "#e5dbff",
    "#d0bfff",
    "#b197fc",
    "#9775fa",
    "#845ef7",
    "#7950f2",
    "#7048e8",
    "#6741d9",
    "#5f3dc4"
  ],
  indigo: [
    "#edf2ff",
    "#dbe4ff",
    "#bac8ff",
    "#91a7ff",
    "#748ffc",
    "#5c7cfa",
    "#4c6ef5",
    "#4263eb",
    "#3b5bdb",
    "#364fc7"
  ],
  blue: [
    "#e7f5ff",
    "#d0ebff",
    "#a5d8ff",
    "#74c0fc",
    "#4dabf7",
    "#339af0",
    "#228be6",
    "#1c7ed6",
    "#1971c2",
    "#1864ab"
  ],
  cyan: [
    "#e3fafc",
    "#c5f6fa",
    "#99e9f2",
    "#66d9e8",
    "#3bc9db",
    "#22b8cf",
    "#15aabf",
    "#1098ad",
    "#0c8599",
    "#0b7285"
  ],
  teal: [
    "#e6fcf5",
    "#c3fae8",
    "#96f2d7",
    "#63e6be",
    "#38d9a9",
    "#20c997",
    "#12b886",
    "#0ca678",
    "#099268",
    "#087f5b"
  ],
  green: [
    "#ebfbee",
    "#d3f9d8",
    "#b2f2bb",
    "#8ce99a",
    "#69db7c",
    "#51cf66",
    "#40c057",
    "#37b24d",
    "#2f9e44",
    "#2b8a3e"
  ],
  lime: [
    "#f4fce3",
    "#e9fac8",
    "#d8f5a2",
    "#c0eb75",
    "#a9e34b",
    "#94d82d",
    "#82c91e",
    "#74b816",
    "#66a80f",
    "#5c940d"
  ],
  yellow: [
    "#fff9db",
    "#fff3bf",
    "#ffec99",
    "#ffe066",
    "#ffd43b",
    "#fcc419",
    "#fab005",
    "#f59f00",
    "#f08c00",
    "#e67700"
  ],
  orange: [
    "#fff4e6",
    "#ffe8cc",
    "#ffd8a8",
    "#ffc078",
    "#ffa94d",
    "#ff922b",
    "#fd7e14",
    "#f76707",
    "#e8590c",
    "#d9480f"
  ]
}, ur = "-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji", rr = {
  scale: 1,
  fontSmoothing: !0,
  focusRing: "auto",
  white: "#fff",
  black: "#000",
  colors: Me,
  primaryShade: { light: 6, dark: 8 },
  primaryColor: "blue",
  variantColorResolver: Te,
  autoContrast: !1,
  luminanceThreshold: 0.3,
  fontFamily: ur,
  fontFamilyMonospace: "ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, Liberation Mono, Courier New, monospace",
  respectReducedMotion: !1,
  cursorType: "default",
  defaultGradient: { from: "blue", to: "cyan", deg: 45 },
  defaultRadius: "sm",
  activeClassName: "mantine-active",
  focusClassName: "",
  headings: {
    fontFamily: ur,
    fontWeight: "700",
    textWrap: "wrap",
    sizes: {
      h1: { fontSize: c(34), lineHeight: "1.3" },
      h2: { fontSize: c(26), lineHeight: "1.35" },
      h3: { fontSize: c(22), lineHeight: "1.4" },
      h4: { fontSize: c(18), lineHeight: "1.45" },
      h5: { fontSize: c(16), lineHeight: "1.5" },
      h6: { fontSize: c(14), lineHeight: "1.5" }
    }
  },
  fontSizes: {
    xs: c(12),
    sm: c(14),
    md: c(16),
    lg: c(18),
    xl: c(20)
  },
  lineHeights: {
    xs: "1.4",
    sm: "1.45",
    md: "1.55",
    lg: "1.6",
    xl: "1.65"
  },
  radius: {
    xs: c(2),
    sm: c(4),
    md: c(8),
    lg: c(16),
    xl: c(32)
  },
  spacing: {
    xs: c(10),
    sm: c(12),
    md: c(16),
    lg: c(20),
    xl: c(32)
  },
  breakpoints: {
    xs: "36em",
    sm: "48em",
    md: "62em",
    lg: "75em",
    xl: "88em"
  },
  shadows: {
    xs: `0 ${c(1)} ${c(3)} rgba(0, 0, 0, 0.05), 0 ${c(1)} ${c(2)} rgba(0, 0, 0, 0.1)`,
    sm: `0 ${c(1)} ${c(3)} rgba(0, 0, 0, 0.05), rgba(0, 0, 0, 0.05) 0 ${c(10)} ${c(
      15
    )} ${c(-5)}, rgba(0, 0, 0, 0.04) 0 ${c(7)} ${c(7)} ${c(-5)}`,
    md: `0 ${c(1)} ${c(3)} rgba(0, 0, 0, 0.05), rgba(0, 0, 0, 0.05) 0 ${c(20)} ${c(
      25
    )} ${c(-5)}, rgba(0, 0, 0, 0.04) 0 ${c(10)} ${c(10)} ${c(-5)}`,
    lg: `0 ${c(1)} ${c(3)} rgba(0, 0, 0, 0.05), rgba(0, 0, 0, 0.05) 0 ${c(28)} ${c(
      23
    )} ${c(-7)}, rgba(0, 0, 0, 0.04) 0 ${c(12)} ${c(12)} ${c(-7)}`,
    xl: `0 ${c(1)} ${c(3)} rgba(0, 0, 0, 0.05), rgba(0, 0, 0, 0.05) 0 ${c(36)} ${c(
      28
    )} ${c(-7)}, rgba(0, 0, 0, 0.04) 0 ${c(17)} ${c(17)} ${c(-7)}`
  },
  other: {},
  components: {}
};
function fr(t) {
  return t === "auto" || t === "dark" || t === "light";
}
function Ee({
  key: t = "mantine-color-scheme-value"
} = {}) {
  let r;
  return {
    get: (e) => {
      if (typeof window > "u")
        return e;
      try {
        const n = window.localStorage.getItem(t);
        return fr(n) ? n : e;
      } catch {
        return e;
      }
    },
    set: (e) => {
      try {
        window.localStorage.setItem(t, e);
      } catch (n) {
        console.warn(
          "[@mantine/core] Local storage color scheme manager was unable to save color scheme.",
          n
        );
      }
    },
    subscribe: (e) => {
      r = (n) => {
        n.storageArea === window.localStorage && n.key === t && fr(n.newValue) && e(n.newValue);
      }, window.addEventListener("storage", r);
    },
    unsubscribe: () => {
      window.removeEventListener("storage", r);
    },
    clear: () => {
      window.localStorage.removeItem(t);
    }
  };
}
const Pe = "[@mantine/core] MantineProvider: Invalid theme.primaryColor, it accepts only key of theme.colors, learn more – https://mantine.dev/theming/colors/#primary-color", pr = "[@mantine/core] MantineProvider: Invalid theme.primaryShade, it accepts only 0-9 integers or an object { light: 0-9, dark: 0-9 }";
function Ot(t) {
  return t < 0 || t > 9 ? !1 : parseInt(t.toString(), 10) === t;
}
function mr(t) {
  if (!(t.primaryColor in t.colors))
    throw new Error(Pe);
  if (typeof t.primaryShade == "object" && (!Ot(t.primaryShade.dark) || !Ot(t.primaryShade.light)))
    throw new Error(pr);
  if (typeof t.primaryShade == "number" && !Ot(t.primaryShade))
    throw new Error(pr);
}
function Le(t, r) {
  var n;
  if (!r)
    return mr(t), t;
  const e = Ut(t, r);
  return r.fontFamily && !((n = r.headings) != null && n.fontFamily) && (e.headings.fontFamily = r.fontFamily), mr(e), e;
}
const er = Xt(null), Ae = () => ht(er) || rr;
function q() {
  const t = ht(er);
  if (!t)
    throw new Error(
      "@mantine/core: MantineProvider was not found in component tree, make sure you have it in your app"
    );
  return t;
}
function xr({
  theme: t,
  children: r,
  inherit: e = !0
}) {
  const n = Ae(), a = Zr(
    () => Le(e ? n : rr, t),
    [t, n, e]
  );
  return /* @__PURE__ */ g(er.Provider, { value: a, children: r });
}
xr.displayName = "@mantine/core/MantineThemeProvider";
function Oe() {
  const t = q(), r = Jt(), e = P(t.breakpoints).reduce((n, a) => {
    const o = t.breakpoints[a].includes("px"), i = ae(t.breakpoints[a]), s = o ? `${i - 0.1}px` : lr(i - 0.1), l = o ? `${i}px` : lr(i);
    return `${n}@media (max-width: ${s}) {.mantine-visible-from-${a} {display: none !important;}}@media (min-width: ${l}) {.mantine-hidden-from-${a} {display: none !important;}}`;
  }, "");
  return /* @__PURE__ */ g(
    "style",
    {
      "data-mantine-styles": "classes",
      nonce: r == null ? void 0 : r(),
      dangerouslySetInnerHTML: { __html: e }
    }
  );
}
function zt(t) {
  return Object.entries(t).map(([r, e]) => `${r}: ${e};`).join("");
}
function Wt(t, r) {
  return (Array.isArray(t) ? t : [t]).reduce((n, a) => `${a}{${n}}`, r);
}
function ze(t, r) {
  const e = zt(t.variables), n = e ? Wt(r, e) : "", a = zt(t.dark), o = a ? Wt(`${r}[data-mantine-color-scheme="dark"]`, a) : "", i = zt(t.light), s = i ? Wt(`${r}[data-mantine-color-scheme="light"]`, i) : "";
  return `${n}${o}${s}`;
}
function We({ color: t, theme: r, autoContrast: e }) {
  return (typeof e == "boolean" ? e : r.autoContrast) && ft({ color: t || r.primaryColor, theme: r }).isLight ? "var(--mantine-color-black)" : "var(--mantine-color-white)";
}
function yr(t, r) {
  return We({
    color: t.colors[t.primaryColor][ut(t, r)],
    theme: t,
    autoContrast: null
  });
}
function yt({
  theme: t,
  color: r,
  colorScheme: e,
  name: n = r,
  withColorValues: a = !0
}) {
  if (!t.colors[r])
    return {};
  if (e === "light") {
    const s = ut(t, "light"), l = {
      [`--mantine-color-${n}-text`]: `var(--mantine-color-${n}-filled)`,
      [`--mantine-color-${n}-filled`]: `var(--mantine-color-${n}-${s})`,
      [`--mantine-color-${n}-filled-hover`]: `var(--mantine-color-${n}-${s === 9 ? 8 : s + 1})`,
      [`--mantine-color-${n}-light`]: et(t.colors[r][s], 0.1),
      [`--mantine-color-${n}-light-hover`]: et(t.colors[r][s], 0.12),
      [`--mantine-color-${n}-light-color`]: `var(--mantine-color-${n}-${s})`,
      [`--mantine-color-${n}-outline`]: `var(--mantine-color-${n}-${s})`,
      [`--mantine-color-${n}-outline-hover`]: et(t.colors[r][s], 0.05)
    };
    return a ? {
      [`--mantine-color-${n}-0`]: t.colors[r][0],
      [`--mantine-color-${n}-1`]: t.colors[r][1],
      [`--mantine-color-${n}-2`]: t.colors[r][2],
      [`--mantine-color-${n}-3`]: t.colors[r][3],
      [`--mantine-color-${n}-4`]: t.colors[r][4],
      [`--mantine-color-${n}-5`]: t.colors[r][5],
      [`--mantine-color-${n}-6`]: t.colors[r][6],
      [`--mantine-color-${n}-7`]: t.colors[r][7],
      [`--mantine-color-${n}-8`]: t.colors[r][8],
      [`--mantine-color-${n}-9`]: t.colors[r][9],
      ...l
    } : l;
  }
  const o = ut(t, "dark"), i = {
    [`--mantine-color-${n}-text`]: `var(--mantine-color-${n}-4)`,
    [`--mantine-color-${n}-filled`]: `var(--mantine-color-${n}-${o})`,
    [`--mantine-color-${n}-filled-hover`]: `var(--mantine-color-${n}-${o === 9 ? 8 : o + 1})`,
    [`--mantine-color-${n}-light`]: et(
      t.colors[r][Math.max(0, o - 2)],
      0.15
    ),
    [`--mantine-color-${n}-light-hover`]: et(
      t.colors[r][Math.max(0, o - 2)],
      0.2
    ),
    [`--mantine-color-${n}-light-color`]: `var(--mantine-color-${n}-${Math.max(o - 5, 0)})`,
    [`--mantine-color-${n}-outline`]: `var(--mantine-color-${n}-${Math.max(o - 4, 0)})`,
    [`--mantine-color-${n}-outline-hover`]: et(
      t.colors[r][Math.max(o - 4, 0)],
      0.05
    )
  };
  return a ? {
    [`--mantine-color-${n}-0`]: t.colors[r][0],
    [`--mantine-color-${n}-1`]: t.colors[r][1],
    [`--mantine-color-${n}-2`]: t.colors[r][2],
    [`--mantine-color-${n}-3`]: t.colors[r][3],
    [`--mantine-color-${n}-4`]: t.colors[r][4],
    [`--mantine-color-${n}-5`]: t.colors[r][5],
    [`--mantine-color-${n}-6`]: t.colors[r][6],
    [`--mantine-color-${n}-7`]: t.colors[r][7],
    [`--mantine-color-${n}-8`]: t.colors[r][8],
    [`--mantine-color-${n}-9`]: t.colors[r][9],
    ...i
  } : i;
}
function Be(t) {
  return Array.isArray(t) ? t : Array(10).fill(t);
}
function Fe(t) {
  return !!t && typeof t == "object" && "mantine-virtual-color" in t;
}
function nt(t, r, e) {
  P(r).forEach(
    (n) => Object.assign(t, { [`--mantine-${e}-${n}`]: r[n] })
  );
}
const Rr = (t) => {
  const r = ut(t, "light"), e = t.defaultRadius in t.radius ? t.radius[t.defaultRadius] : c(t.defaultRadius), n = {
    variables: {
      "--mantine-scale": t.scale.toString(),
      "--mantine-cursor-type": t.cursorType,
      "--mantine-color-scheme": "light dark",
      "--mantine-webkit-font-smoothing": t.fontSmoothing ? "antialiased" : "unset",
      "--mantine-moz-font-smoothing": t.fontSmoothing ? "grayscale" : "unset",
      "--mantine-color-white": t.white,
      "--mantine-color-black": t.black,
      "--mantine-line-height": t.lineHeights.md,
      "--mantine-font-family": t.fontFamily,
      "--mantine-font-family-monospace": t.fontFamilyMonospace,
      "--mantine-font-family-headings": t.headings.fontFamily,
      "--mantine-heading-font-weight": t.headings.fontWeight,
      "--mantine-heading-text-wrap": t.headings.textWrap,
      "--mantine-radius-default": e,
      // Primary colors
      "--mantine-primary-color-filled": `var(--mantine-color-${t.primaryColor}-filled)`,
      "--mantine-primary-color-filled-hover": `var(--mantine-color-${t.primaryColor}-filled-hover)`,
      "--mantine-primary-color-light": `var(--mantine-color-${t.primaryColor}-light)`,
      "--mantine-primary-color-light-hover": `var(--mantine-color-${t.primaryColor}-light-hover)`,
      "--mantine-primary-color-light-color": `var(--mantine-color-${t.primaryColor}-light-color)`
    },
    light: {
      "--mantine-primary-color-contrast": yr(t, "light"),
      "--mantine-color-bright": "var(--mantine-color-black)",
      "--mantine-color-text": t.black,
      "--mantine-color-body": t.white,
      "--mantine-color-error": "var(--mantine-color-red-6)",
      "--mantine-color-placeholder": "var(--mantine-color-gray-5)",
      "--mantine-color-anchor": `var(--mantine-color-${t.primaryColor}-${r})`,
      "--mantine-color-default": "var(--mantine-color-white)",
      "--mantine-color-default-hover": "var(--mantine-color-gray-0)",
      "--mantine-color-default-color": "var(--mantine-color-black)",
      "--mantine-color-default-border": "var(--mantine-color-gray-4)",
      "--mantine-color-dimmed": "var(--mantine-color-gray-6)"
    },
    dark: {
      "--mantine-primary-color-contrast": yr(t, "dark"),
      "--mantine-color-bright": "var(--mantine-color-white)",
      "--mantine-color-text": "var(--mantine-color-dark-0)",
      "--mantine-color-body": "var(--mantine-color-dark-7)",
      "--mantine-color-error": "var(--mantine-color-red-8)",
      "--mantine-color-placeholder": "var(--mantine-color-dark-3)",
      "--mantine-color-anchor": `var(--mantine-color-${t.primaryColor}-4)`,
      "--mantine-color-default": "var(--mantine-color-dark-6)",
      "--mantine-color-default-hover": "var(--mantine-color-dark-5)",
      "--mantine-color-default-color": "var(--mantine-color-white)",
      "--mantine-color-default-border": "var(--mantine-color-dark-4)",
      "--mantine-color-dimmed": "var(--mantine-color-dark-2)"
    }
  };
  nt(n.variables, t.breakpoints, "breakpoint"), nt(n.variables, t.spacing, "spacing"), nt(n.variables, t.fontSizes, "font-size"), nt(n.variables, t.lineHeights, "line-height"), nt(n.variables, t.shadows, "shadow"), nt(n.variables, t.radius, "radius"), t.colors[t.primaryColor].forEach((o, i) => {
    n.variables[`--mantine-primary-color-${i}`] = `var(--mantine-color-${t.primaryColor}-${i})`;
  }), P(t.colors).forEach((o) => {
    const i = t.colors[o];
    if (Fe(i)) {
      Object.assign(
        n.light,
        yt({
          theme: t,
          name: i.name,
          color: i.light,
          colorScheme: "light",
          withColorValues: !0
        })
      ), Object.assign(
        n.dark,
        yt({
          theme: t,
          name: i.name,
          color: i.dark,
          colorScheme: "dark",
          withColorValues: !0
        })
      );
      return;
    }
    i.forEach((s, l) => {
      n.variables[`--mantine-color-${o}-${l}`] = s;
    }), Object.assign(
      n.light,
      yt({
        theme: t,
        color: o,
        colorScheme: "light",
        withColorValues: !1
      })
    ), Object.assign(
      n.dark,
      yt({
        theme: t,
        color: o,
        colorScheme: "dark",
        withColorValues: !1
      })
    );
  });
  const a = t.headings.sizes;
  return P(a).forEach((o) => {
    n.variables[`--mantine-${o}-font-size`] = a[o].fontSize, n.variables[`--mantine-${o}-line-height`] = a[o].lineHeight, n.variables[`--mantine-${o}-font-weight`] = a[o].fontWeight || t.headings.fontWeight;
  }), n;
};
function je({ theme: t, generator: r }) {
  const e = Rr(t), n = r == null ? void 0 : r(t);
  return n ? Ut(e, n) : e;
}
const Bt = Rr(rr);
function Ve(t) {
  const r = {
    variables: {},
    light: {},
    dark: {}
  };
  return P(t.variables).forEach((e) => {
    Bt.variables[e] !== t.variables[e] && (r.variables[e] = t.variables[e]);
  }), P(t.light).forEach((e) => {
    Bt.light[e] !== t.light[e] && (r.light[e] = t.light[e]);
  }), P(t.dark).forEach((e) => {
    Bt.dark[e] !== t.dark[e] && (r.dark[e] = t.dark[e]);
  }), r;
}
function He(t) {
  return `
  ${t}[data-mantine-color-scheme="dark"] { --mantine-color-scheme: dark; }
  ${t}[data-mantine-color-scheme="light"] { --mantine-color-scheme: light; }
`;
}
function Tr({
  cssVariablesSelector: t,
  deduplicateCssVariables: r
}) {
  const e = q(), n = Jt(), a = he(), o = je({ theme: e, generator: a }), i = t === ":root" && r, s = i ? Ve(o) : o, l = ze(s, t);
  return l ? /* @__PURE__ */ g(
    "style",
    {
      "data-mantine-styles": !0,
      nonce: n == null ? void 0 : n(),
      dangerouslySetInnerHTML: {
        __html: `${l}${i ? "" : He(t)}`
      }
    }
  ) : null;
}
Tr.displayName = "@mantine/CssVariables";
function De() {
  const t = console.error;
  console.error = (...r) => {
    r.length > 1 && typeof r[0] == "string" && r[0].toLowerCase().includes("extra attributes from the server") && typeof r[1] == "string" && r[1].toLowerCase().includes("data-mantine-color-scheme") || t(...r);
  };
}
function ot(t, r) {
  var n;
  const e = t !== "auto" ? t : window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light";
  (n = r()) == null || n.setAttribute("data-mantine-color-scheme", e);
}
function Ye({
  manager: t,
  defaultColorScheme: r,
  getRootElement: e,
  forceColorScheme: n
}) {
  const a = at(), [o, i] = Y(() => t.get(r)), s = n || o, l = bt(
    (u) => {
      n || (ot(u, e), i(u), t.set(u));
    },
    [t.set, s, n]
  ), d = bt(() => {
    i(r), ot(r, e), t.clear();
  }, [t.clear, r]);
  return B(() => (t.subscribe(l), t.unsubscribe), [t.subscribe, t.unsubscribe]), Kt(() => {
    ot(t.get(r), e);
  }, []), B(() => {
    var f;
    if (n)
      return ot(n, e), () => {
      };
    n === void 0 && ot(o, e), a.current = window.matchMedia("(prefers-color-scheme: dark)");
    const u = (p) => {
      o === "auto" && ot(p.matches ? "dark" : "light", e);
    };
    return (f = a.current) == null || f.addEventListener("change", u), () => {
      var p;
      return (p = a.current) == null ? void 0 : p.removeEventListener("change", u);
    };
  }, [o, n]), { colorScheme: s, setColorScheme: l, clearColorScheme: d };
}
function Ge({
  respectReducedMotion: t,
  getRootElement: r
}) {
  Kt(() => {
    var e;
    t && ((e = r()) == null || e.setAttribute("data-respect-reduced-motion", "true"));
  }, [t]);
}
De();
function Mr({
  theme: t,
  children: r,
  getStyleNonce: e,
  withStaticClasses: n = !0,
  withGlobalClasses: a = !0,
  deduplicateCssVariables: o = !0,
  withCssVariables: i = !0,
  cssVariablesSelector: s = ":root",
  classNamesPrefix: l = "mantine",
  colorSchemeManager: d = Ee(),
  defaultColorScheme: u = "light",
  getRootElement: f = () => document.documentElement,
  cssVariablesResolver: p,
  forceColorScheme: y,
  stylesTransform: m
}) {
  const { colorScheme: b, setColorScheme: v, clearColorScheme: w } = Ye({
    defaultColorScheme: u,
    forceColorScheme: y,
    manager: d,
    getRootElement: f
  });
  return Ge({
    respectReducedMotion: (t == null ? void 0 : t.respectReducedMotion) || !1,
    getRootElement: f
  }), /* @__PURE__ */ g(
    Ir.Provider,
    {
      value: {
        colorScheme: b,
        setColorScheme: v,
        clearColorScheme: w,
        getRootElement: f,
        classNamesPrefix: l,
        getStyleNonce: e,
        cssVariablesResolver: p,
        cssVariablesSelector: s,
        withStaticClasses: n,
        stylesTransform: m
      },
      children: /* @__PURE__ */ F(xr, { theme: t, children: [
        i && /* @__PURE__ */ g(
          Tr,
          {
            cssVariablesSelector: s,
            deduplicateCssVariables: o
          }
        ),
        a && /* @__PURE__ */ g(Oe, {}),
        r
      ] })
    }
  );
}
Mr.displayName = "@mantine/core/MantineProvider";
const Xe = {
  always: "mantine-focus-always",
  auto: "mantine-focus-auto",
  never: "mantine-focus-never"
};
function Ue({ theme: t, options: r, unstyled: e }) {
  return X(
    (r == null ? void 0 : r.focusable) && !e && (t.focusClassName || Xe[t.focusRing]),
    (r == null ? void 0 : r.active) && !e && t.activeClassName
  );
}
function qe({
  selector: t,
  stylesCtx: r,
  options: e,
  props: n,
  theme: a
}) {
  return Zt({
    theme: a,
    classNames: e == null ? void 0 : e.classNames,
    props: (e == null ? void 0 : e.props) || n,
    stylesCtx: r
  })[t];
}
function gr({
  selector: t,
  stylesCtx: r,
  theme: e,
  classNames: n,
  props: a
}) {
  return Zt({ theme: e, classNames: n, props: a, stylesCtx: r })[t];
}
function Qe({ rootSelector: t, selector: r, className: e }) {
  return t === r ? e : void 0;
}
function Ke({ selector: t, classes: r, unstyled: e }) {
  return e ? void 0 : r[t];
}
function Ze({
  themeName: t,
  classNamesPrefix: r,
  selector: e,
  withStaticClass: n
}) {
  return n === !1 ? [] : t.map((a) => `${r}-${a}-${e}`);
}
function Je({
  themeName: t,
  theme: r,
  selector: e,
  props: n,
  stylesCtx: a
}) {
  return t.map(
    (o) => {
      var i, s;
      return (s = Zt({
        theme: r,
        classNames: (i = r.components[o]) == null ? void 0 : i.classNames,
        props: n,
        stylesCtx: a
      })) == null ? void 0 : s[e];
    }
  );
}
function tn({
  options: t,
  classes: r,
  selector: e,
  unstyled: n
}) {
  return t != null && t.variant && !n ? r[`${e}--${t.variant}`] : void 0;
}
function rn({
  theme: t,
  options: r,
  themeName: e,
  selector: n,
  classNamesPrefix: a,
  classNames: o,
  classes: i,
  unstyled: s,
  className: l,
  rootSelector: d,
  props: u,
  stylesCtx: f,
  withStaticClasses: p,
  headless: y,
  transformedStyles: m
}) {
  return X(
    Ue({ theme: t, options: r, unstyled: s || y }),
    Je({ theme: t, themeName: e, selector: n, props: u, stylesCtx: f }),
    tn({ options: r, classes: i, selector: n, unstyled: s }),
    gr({ selector: n, stylesCtx: f, theme: t, classNames: o, props: u }),
    gr({ selector: n, stylesCtx: f, theme: t, classNames: m, props: u }),
    qe({ selector: n, stylesCtx: f, options: r, props: u, theme: t }),
    Qe({ rootSelector: d, selector: n, className: l }),
    Ke({ selector: n, classes: i, unstyled: s || y }),
    p && !y && Ze({
      themeName: e,
      classNamesPrefix: a,
      selector: n,
      withStaticClass: r == null ? void 0 : r.withStaticClass
    }),
    r == null ? void 0 : r.className
  );
}
function en({
  theme: t,
  themeName: r,
  props: e,
  stylesCtx: n,
  selector: a
}) {
  return r.map(
    (o) => {
      var i;
      return Ht({
        theme: t,
        styles: (i = t.components[o]) == null ? void 0 : i.styles,
        props: e,
        stylesCtx: n
      })[a];
    }
  ).reduce((o, i) => ({ ...o, ...i }), {});
}
function Yt({ style: t, theme: r }) {
  return Array.isArray(t) ? [...t].reduce(
    (e, n) => ({ ...e, ...Yt({ style: n, theme: r }) }),
    {}
  ) : typeof t == "function" ? t(r) : t ?? {};
}
function nn(t) {
  return t.reduce((r, e) => (e && Object.keys(e).forEach((n) => {
    r[n] = { ...r[n], ...qt(e[n]) };
  }), r), {});
}
function on({
  vars: t,
  varsResolver: r,
  theme: e,
  props: n,
  stylesCtx: a,
  selector: o,
  themeName: i,
  headless: s
}) {
  var l;
  return (l = nn([
    s ? {} : r == null ? void 0 : r(e, n, a),
    ...i.map((d) => {
      var u, f, p;
      return (p = (f = (u = e.components) == null ? void 0 : u[d]) == null ? void 0 : f.vars) == null ? void 0 : p.call(f, e, n, a);
    }),
    t == null ? void 0 : t(e, n, a)
  ])) == null ? void 0 : l[o];
}
function an({
  theme: t,
  themeName: r,
  selector: e,
  options: n,
  props: a,
  stylesCtx: o,
  rootSelector: i,
  styles: s,
  style: l,
  vars: d,
  varsResolver: u,
  headless: f,
  withStylesTransform: p
}) {
  return {
    ...!p && en({ theme: t, themeName: r, props: a, stylesCtx: o, selector: e }),
    ...!p && Ht({ theme: t, styles: s, props: a, stylesCtx: o })[e],
    ...!p && Ht({ theme: t, styles: n == null ? void 0 : n.styles, props: (n == null ? void 0 : n.props) || a, stylesCtx: o })[e],
    ...on({ theme: t, props: a, stylesCtx: o, vars: d, varsResolver: u, selector: e, themeName: r, headless: f }),
    ...i === e ? Yt({ style: l, theme: t }) : null,
    ...Yt({ style: n == null ? void 0 : n.style, theme: t })
  };
}
function sn({ props: t, stylesCtx: r, themeName: e }) {
  var i;
  const n = q(), a = (i = Ce()) == null ? void 0 : i();
  return {
    getTransformedStyles: (s) => a ? [
      ...s.map(
        (d) => a(d, { props: t, theme: n, ctx: r })
      ),
      ...e.map(
        (d) => {
          var u;
          return a((u = n.components[d]) == null ? void 0 : u.styles, { props: t, theme: n, ctx: r });
        }
      )
    ].filter(Boolean) : [],
    withStylesTransform: !!a
  };
}
function E({
  name: t,
  classes: r,
  props: e,
  stylesCtx: n,
  className: a,
  style: o,
  rootSelector: i = "root",
  unstyled: s,
  classNames: l,
  styles: d,
  vars: u,
  varsResolver: f
}) {
  const p = q(), y = ve(), m = $e(), b = Se(), v = (Array.isArray(t) ? t : [t]).filter((h) => h), { withStylesTransform: w, getTransformedStyles: $ } = sn({
    props: e,
    stylesCtx: n,
    themeName: v
  });
  return (h, S) => ({
    className: rn({
      theme: p,
      options: S,
      themeName: v,
      selector: h,
      classNamesPrefix: y,
      classNames: l,
      classes: r,
      unstyled: s,
      className: a,
      rootSelector: i,
      props: e,
      stylesCtx: n,
      withStaticClasses: m,
      headless: b,
      transformedStyles: $([S == null ? void 0 : S.styles, d])
    }),
    style: an({
      theme: p,
      themeName: v,
      selector: h,
      options: S,
      props: e,
      stylesCtx: n,
      rootSelector: i,
      styles: d,
      style: o,
      vars: u,
      varsResolver: f,
      headless: b,
      withStylesTransform: w
    })
  });
}
function I(t, r, e) {
  var i;
  const n = q(), a = (i = n.components[t]) == null ? void 0 : i.defaultProps, o = typeof a == "function" ? a(n) : a;
  return { ...r, ...o, ...qt(e) };
}
function Ft(t) {
  return P(t).reduce(
    (r, e) => t[e] !== void 0 ? `${r}${ne(e)}:${t[e]};` : r,
    ""
  ).trim();
}
function cn({ selector: t, styles: r, media: e, container: n }) {
  const a = r ? Ft(r) : "", o = Array.isArray(e) ? e.map((s) => `@media${s.query}{${t}{${Ft(s.styles)}}}`) : [], i = Array.isArray(n) ? n.map(
    (s) => `@container ${s.query}{${t}{${Ft(s.styles)}}}`
  ) : [];
  return `${a ? `${t}{${a}}` : ""}${o.join("")}${i.join("")}`.trim();
}
function ln(t) {
  const r = Jt();
  return /* @__PURE__ */ g(
    "style",
    {
      "data-mantine-styles": "inline",
      nonce: r == null ? void 0 : r(),
      dangerouslySetInnerHTML: { __html: cn(t) }
    }
  );
}
function Er(t) {
  const {
    m: r,
    mx: e,
    my: n,
    mt: a,
    mb: o,
    ml: i,
    mr: s,
    me: l,
    ms: d,
    p: u,
    px: f,
    py: p,
    pt: y,
    pb: m,
    pl: b,
    pr: v,
    pe: w,
    ps: $,
    bg: h,
    c: S,
    opacity: _,
    ff: k,
    fz: T,
    fw: x,
    lts: V,
    ta: H,
    lh: K,
    fs: Z,
    tt: L,
    td: A,
    w: st,
    miw: J,
    maw: tt,
    h: rt,
    mih: N,
    mah: D,
    bgsz: O,
    bgp: ct,
    bgr: pt,
    bga: It,
    pos: xt,
    top: Rt,
    left: Tt,
    bottom: Mt,
    right: Et,
    inset: jr,
    display: Vr,
    flex: Hr,
    hiddenFrom: Dr,
    visibleFrom: Yr,
    lightHidden: Gr,
    darkHidden: Xr,
    sx: Ur,
    ...qr
  } = t;
  return { styleProps: qt({
    m: r,
    mx: e,
    my: n,
    mt: a,
    mb: o,
    ml: i,
    mr: s,
    me: l,
    ms: d,
    p: u,
    px: f,
    py: p,
    pt: y,
    pb: m,
    pl: b,
    pr: v,
    pe: w,
    ps: $,
    bg: h,
    c: S,
    opacity: _,
    ff: k,
    fz: T,
    fw: x,
    lts: V,
    ta: H,
    lh: K,
    fs: Z,
    tt: L,
    td: A,
    w: st,
    miw: J,
    maw: tt,
    h: rt,
    mih: N,
    mah: D,
    bgsz: O,
    bgp: ct,
    bgr: pt,
    bga: It,
    pos: xt,
    top: Rt,
    left: Tt,
    bottom: Mt,
    right: Et,
    inset: jr,
    display: Vr,
    flex: Hr,
    hiddenFrom: Dr,
    visibleFrom: Yr,
    lightHidden: Gr,
    darkHidden: Xr,
    sx: Ur
  }), rest: qr };
}
const dn = {
  m: { type: "spacing", property: "margin" },
  mt: { type: "spacing", property: "marginTop" },
  mb: { type: "spacing", property: "marginBottom" },
  ml: { type: "spacing", property: "marginLeft" },
  mr: { type: "spacing", property: "marginRight" },
  ms: { type: "spacing", property: "marginInlineStart" },
  me: { type: "spacing", property: "marginInlineEnd" },
  mx: { type: "spacing", property: "marginInline" },
  my: { type: "spacing", property: "marginBlock" },
  p: { type: "spacing", property: "padding" },
  pt: { type: "spacing", property: "paddingTop" },
  pb: { type: "spacing", property: "paddingBottom" },
  pl: { type: "spacing", property: "paddingLeft" },
  pr: { type: "spacing", property: "paddingRight" },
  ps: { type: "spacing", property: "paddingInlineStart" },
  pe: { type: "spacing", property: "paddingInlineEnd" },
  px: { type: "spacing", property: "paddingInline" },
  py: { type: "spacing", property: "paddingBlock" },
  bd: { type: "border", property: "border" },
  bg: { type: "color", property: "background" },
  c: { type: "textColor", property: "color" },
  opacity: { type: "identity", property: "opacity" },
  ff: { type: "fontFamily", property: "fontFamily" },
  fz: { type: "fontSize", property: "fontSize" },
  fw: { type: "identity", property: "fontWeight" },
  lts: { type: "size", property: "letterSpacing" },
  ta: { type: "identity", property: "textAlign" },
  lh: { type: "lineHeight", property: "lineHeight" },
  fs: { type: "identity", property: "fontStyle" },
  tt: { type: "identity", property: "textTransform" },
  td: { type: "identity", property: "textDecoration" },
  w: { type: "spacing", property: "width" },
  miw: { type: "spacing", property: "minWidth" },
  maw: { type: "spacing", property: "maxWidth" },
  h: { type: "spacing", property: "height" },
  mih: { type: "spacing", property: "minHeight" },
  mah: { type: "spacing", property: "maxHeight" },
  bgsz: { type: "size", property: "backgroundSize" },
  bgp: { type: "identity", property: "backgroundPosition" },
  bgr: { type: "identity", property: "backgroundRepeat" },
  bga: { type: "identity", property: "backgroundAttachment" },
  pos: { type: "identity", property: "position" },
  top: { type: "identity", property: "top" },
  left: { type: "size", property: "left" },
  bottom: { type: "size", property: "bottom" },
  right: { type: "size", property: "right" },
  inset: { type: "size", property: "inset" },
  display: { type: "identity", property: "display" },
  flex: { type: "identity", property: "flex" }
};
function nr(t, r) {
  const e = ft({ color: t, theme: r });
  return e.color === "dimmed" ? "var(--mantine-color-dimmed)" : e.color === "bright" ? "var(--mantine-color-bright)" : e.variable ? `var(${e.variable})` : e.color;
}
function un(t, r) {
  const e = ft({ color: t, theme: r });
  return e.isThemeColor && e.shade === void 0 ? `var(--mantine-color-${e.color}-text)` : nr(t, r);
}
function fn(t, r) {
  if (typeof t == "number")
    return c(t);
  if (typeof t == "string") {
    const [e, n, ...a] = t.split(" ").filter((i) => i.trim() !== "");
    let o = `${c(e)}`;
    return n && (o += ` ${n}`), Be.length > 0 && (o += ` ${nr(a.join(" "), r)}`), o.trim();
  }
  return t;
}
const br = {
  text: "var(--mantine-font-family)",
  mono: "var(--mantine-font-family-monospace)",
  monospace: "var(--mantine-font-family-monospace)",
  heading: "var(--mantine-font-family-headings)",
  headings: "var(--mantine-font-family-headings)"
};
function pn(t) {
  return typeof t == "string" && t in br ? br[t] : t;
}
const mn = ["h1", "h2", "h3", "h4", "h5", "h6"];
function yn(t, r) {
  return typeof t == "string" && t in r.fontSizes ? `var(--mantine-font-size-${t})` : typeof t == "string" && mn.includes(t) ? `var(--mantine-${t}-font-size)` : typeof t == "number" || typeof t == "string" ? c(t) : t;
}
function gn(t) {
  return t;
}
const bn = ["h1", "h2", "h3", "h4", "h5", "h6"];
function hn(t, r) {
  return typeof t == "string" && t in r.lineHeights ? `var(--mantine-line-height-${t})` : typeof t == "string" && bn.includes(t) ? `var(--mantine-${t}-line-height)` : t;
}
function vn(t) {
  return typeof t == "number" ? c(t) : t;
}
function $n(t, r) {
  if (typeof t == "number")
    return c(t);
  if (typeof t == "string") {
    const e = t.replace("-", "");
    if (!(e in r.spacing))
      return c(t);
    const n = `--mantine-spacing-${e}`;
    return t.startsWith("-") ? `calc(var(${n}) * -1)` : `var(${n})`;
  }
  return t;
}
const jt = {
  color: nr,
  textColor: un,
  fontSize: yn,
  spacing: $n,
  identity: gn,
  size: vn,
  lineHeight: hn,
  fontFamily: pn,
  border: fn
};
function hr(t) {
  return t.replace("(min-width: ", "").replace("em)", "");
}
function Sn({
  media: t,
  ...r
}) {
  const n = Object.keys(t).sort((a, o) => Number(hr(a)) - Number(hr(o))).map((a) => ({ query: a, styles: t[a] }));
  return { ...r, media: n };
}
function wn(t) {
  if (typeof t != "object" || t === null)
    return !1;
  const r = Object.keys(t);
  return !(r.length === 1 && r[0] === "base");
}
function Cn(t) {
  return typeof t == "object" && t !== null ? "base" in t ? t.base : void 0 : t;
}
function _n(t) {
  return typeof t == "object" && t !== null ? P(t).filter((r) => r !== "base") : [];
}
function Nn(t, r) {
  return typeof t == "object" && t !== null && r in t ? t[r] : t;
}
function kn({
  styleProps: t,
  data: r,
  theme: e
}) {
  return Sn(
    P(t).reduce(
      (n, a) => {
        if (a === "hiddenFrom" || a === "visibleFrom" || a === "sx")
          return n;
        const o = r[a], i = Array.isArray(o.property) ? o.property : [o.property], s = Cn(t[a]);
        if (!wn(t[a]))
          return i.forEach((d) => {
            n.inlineStyles[d] = jt[o.type](s, e);
          }), n;
        n.hasResponsiveStyles = !0;
        const l = _n(t[a]);
        return i.forEach((d) => {
          s && (n.styles[d] = jt[o.type](s, e)), l.forEach((u) => {
            const f = `(min-width: ${e.breakpoints[u]})`;
            n.media[f] = {
              ...n.media[f],
              [d]: jt[o.type](
                Nn(t[a], u),
                e
              )
            };
          });
        }), n;
      },
      {
        hasResponsiveStyles: !1,
        styles: {},
        inlineStyles: {},
        media: {}
      }
    )
  );
}
function In() {
  return `__m__-${Jr().replace(/:/g, "")}`;
}
function Pr(t) {
  return t.startsWith("data-") ? t : `data-${t}`;
}
function xn(t) {
  return Object.keys(t).reduce((r, e) => {
    const n = t[e];
    return n === void 0 || n === "" || n === !1 || n === null || (r[Pr(e)] = t[e]), r;
  }, {});
}
function Lr(t) {
  return t ? typeof t == "string" ? { [Pr(t)]: !0 } : Array.isArray(t) ? [...t].reduce(
    (r, e) => ({ ...r, ...Lr(e) }),
    {}
  ) : xn(t) : null;
}
function Gt(t, r) {
  return Array.isArray(t) ? [...t].reduce(
    (e, n) => ({ ...e, ...Gt(n, r) }),
    {}
  ) : typeof t == "function" ? t(r) : t ?? {};
}
function Rn({
  theme: t,
  style: r,
  vars: e,
  styleProps: n
}) {
  const a = Gt(r, t), o = Gt(e, t);
  return { ...a, ...o, ...n };
}
const Ar = it(
  ({
    component: t,
    style: r,
    __vars: e,
    className: n,
    variant: a,
    mod: o,
    size: i,
    hiddenFrom: s,
    visibleFrom: l,
    lightHidden: d,
    darkHidden: u,
    renderRoot: f,
    ...p
  }, y) => {
    var T;
    const m = q(), b = t || "div", { styleProps: v, rest: w } = Er(p), $ = we(), h = (T = $ == null ? void 0 : $()) == null ? void 0 : T(v.sx), S = In(), _ = kn({
      styleProps: v,
      theme: m,
      data: dn
    }), k = {
      ref: y,
      style: Rn({
        theme: m,
        style: r,
        vars: e,
        styleProps: _.inlineStyles
      }),
      className: X(n, h, {
        [S]: _.hasResponsiveStyles,
        "mantine-light-hidden": d,
        "mantine-dark-hidden": u,
        [`mantine-hidden-from-${s}`]: s,
        [`mantine-visible-from-${l}`]: l
      }),
      "data-variant": a,
      "data-size": Nr(i) ? void 0 : i || void 0,
      ...Lr(o),
      ...w
    };
    return /* @__PURE__ */ F(Vt, { children: [
      _.hasResponsiveStyles && /* @__PURE__ */ g(
        ln,
        {
          selector: `.${S}`,
          styles: _.styles,
          media: _.media
        }
      ),
      typeof f == "function" ? f(k) : /* @__PURE__ */ g(b, { ...k })
    ] });
  }
);
Ar.displayName = "@mantine/core/Box";
const C = Ar;
function Or(t) {
  return t;
}
function Q(t) {
  const r = it(t);
  return r.extend = Or, r;
}
function vt(t) {
  const r = it(t);
  return r.extend = Or, r;
}
var zr = { root: "m_87cf2631" };
const Tn = {
  __staticSelector: "UnstyledButton"
}, or = vt(
  (t, r) => {
    const e = I("UnstyledButton", Tn, t), {
      className: n,
      component: a = "button",
      __staticSelector: o,
      unstyled: i,
      classNames: s,
      styles: l,
      style: d,
      ...u
    } = e, f = E({
      name: o,
      props: e,
      classes: zr,
      className: n,
      style: d,
      classNames: s,
      styles: l,
      unstyled: i
    });
    return /* @__PURE__ */ g(
      C,
      {
        ...f("root", { focusable: !0 }),
        component: a,
        ref: r,
        type: a === "button" ? "button" : void 0,
        ...u
      }
    );
  }
);
or.classes = zr;
or.displayName = "@mantine/core/UnstyledButton";
const dt = (t) => ({
  in: { opacity: 1, transform: "scale(1)" },
  out: { opacity: 0, transform: `scale(.9) translateY(${c(t === "bottom" ? 10 : -10)})` },
  transitionProperty: "transform, opacity"
}), gt = {
  fade: {
    in: { opacity: 1 },
    out: { opacity: 0 },
    transitionProperty: "opacity"
  },
  "fade-up": {
    in: { opacity: 1, transform: "translateY(0)" },
    out: { opacity: 0, transform: `translateY(${c(30)}` },
    transitionProperty: "opacity, transform"
  },
  "fade-down": {
    in: { opacity: 1, transform: "translateY(0)" },
    out: { opacity: 0, transform: `translateY(${c(-30)}` },
    transitionProperty: "opacity, transform"
  },
  "fade-left": {
    in: { opacity: 1, transform: "translateX(0)" },
    out: { opacity: 0, transform: `translateX(${c(30)}` },
    transitionProperty: "opacity, transform"
  },
  "fade-right": {
    in: { opacity: 1, transform: "translateX(0)" },
    out: { opacity: 0, transform: `translateX(${c(-30)}` },
    transitionProperty: "opacity, transform"
  },
  scale: {
    in: { opacity: 1, transform: "scale(1)" },
    out: { opacity: 0, transform: "scale(0)" },
    common: { transformOrigin: "top" },
    transitionProperty: "transform, opacity"
  },
  "scale-y": {
    in: { opacity: 1, transform: "scaleY(1)" },
    out: { opacity: 0, transform: "scaleY(0)" },
    common: { transformOrigin: "top" },
    transitionProperty: "transform, opacity"
  },
  "scale-x": {
    in: { opacity: 1, transform: "scaleX(1)" },
    out: { opacity: 0, transform: "scaleX(0)" },
    common: { transformOrigin: "left" },
    transitionProperty: "transform, opacity"
  },
  "skew-up": {
    in: { opacity: 1, transform: "translateY(0) skew(0deg, 0deg)" },
    out: { opacity: 0, transform: `translateY(${c(-20)}) skew(-10deg, -5deg)` },
    common: { transformOrigin: "top" },
    transitionProperty: "transform, opacity"
  },
  "skew-down": {
    in: { opacity: 1, transform: "translateY(0) skew(0deg, 0deg)" },
    out: { opacity: 0, transform: `translateY(${c(20)}) skew(-10deg, -5deg)` },
    common: { transformOrigin: "bottom" },
    transitionProperty: "transform, opacity"
  },
  "rotate-left": {
    in: { opacity: 1, transform: "translateY(0) rotate(0deg)" },
    out: { opacity: 0, transform: `translateY(${c(20)}) rotate(-5deg)` },
    common: { transformOrigin: "bottom" },
    transitionProperty: "transform, opacity"
  },
  "rotate-right": {
    in: { opacity: 1, transform: "translateY(0) rotate(0deg)" },
    out: { opacity: 0, transform: `translateY(${c(20)}) rotate(5deg)` },
    common: { transformOrigin: "top" },
    transitionProperty: "transform, opacity"
  },
  "slide-down": {
    in: { opacity: 1, transform: "translateY(0)" },
    out: { opacity: 0, transform: "translateY(-100%)" },
    common: { transformOrigin: "top" },
    transitionProperty: "transform, opacity"
  },
  "slide-up": {
    in: { opacity: 1, transform: "translateY(0)" },
    out: { opacity: 0, transform: "translateY(100%)" },
    common: { transformOrigin: "bottom" },
    transitionProperty: "transform, opacity"
  },
  "slide-left": {
    in: { opacity: 1, transform: "translateX(0)" },
    out: { opacity: 0, transform: "translateX(100%)" },
    common: { transformOrigin: "left" },
    transitionProperty: "transform, opacity"
  },
  "slide-right": {
    in: { opacity: 1, transform: "translateX(0)" },
    out: { opacity: 0, transform: "translateX(-100%)" },
    common: { transformOrigin: "right" },
    transitionProperty: "transform, opacity"
  },
  pop: {
    ...dt("bottom"),
    common: { transformOrigin: "center center" }
  },
  "pop-bottom-left": {
    ...dt("bottom"),
    common: { transformOrigin: "bottom left" }
  },
  "pop-bottom-right": {
    ...dt("bottom"),
    common: { transformOrigin: "bottom right" }
  },
  "pop-top-left": {
    ...dt("top"),
    common: { transformOrigin: "top left" }
  },
  "pop-top-right": {
    ...dt("top"),
    common: { transformOrigin: "top right" }
  }
}, vr = {
  entering: "in",
  entered: "in",
  exiting: "out",
  exited: "out",
  "pre-exiting": "out",
  "pre-entering": "out"
};
function Mn({
  transition: t,
  state: r,
  duration: e,
  timingFunction: n
}) {
  const a = {
    transitionDuration: `${e}ms`,
    transitionTimingFunction: n
  };
  return typeof t == "string" ? t in gt ? {
    transitionProperty: gt[t].transitionProperty,
    ...a,
    ...gt[t].common,
    ...gt[t][vr[r]]
  } : {} : {
    transitionProperty: t.transitionProperty,
    ...a,
    ...t.common,
    ...t[vr[r]]
  };
}
function En({
  duration: t,
  exitDuration: r,
  timingFunction: e,
  mounted: n,
  onEnter: a,
  onExit: o,
  onEntered: i,
  onExited: s
}) {
  const l = q(), d = ye(), u = l.respectReducedMotion ? d : !1, [f, p] = Y(u ? 0 : t), [y, m] = Y(n ? "entered" : "exited"), b = at(-1), v = at(-1), w = ($) => {
    const h = $ ? a : o, S = $ ? i : s;
    window.clearTimeout(b.current);
    const _ = u ? 0 : $ ? t : r;
    p(_), _ === 0 ? (typeof h == "function" && h(), typeof S == "function" && S(), m($ ? "entered" : "exited")) : v.current = requestAnimationFrame(() => {
      ee.flushSync(() => {
        m($ ? "pre-entering" : "pre-exiting");
      }), v.current = requestAnimationFrame(() => {
        typeof h == "function" && h(), m($ ? "entering" : "exiting"), b.current = window.setTimeout(() => {
          typeof S == "function" && S(), m($ ? "entered" : "exited");
        }, _);
      });
    });
  };
  return ue(() => {
    w(n);
  }, [n]), B(
    () => () => {
      window.clearTimeout(b.current), cancelAnimationFrame(v.current);
    },
    []
  ), {
    transitionDuration: f,
    transitionStatus: y,
    transitionTimingFunction: e || "ease"
  };
}
function Wr({
  keepMounted: t,
  transition: r = "fade",
  duration: e = 250,
  exitDuration: n = e,
  mounted: a,
  children: o,
  timingFunction: i = "ease",
  onExit: s,
  onEntered: l,
  onEnter: d,
  onExited: u
}) {
  const { transitionDuration: f, transitionStatus: p, transitionTimingFunction: y } = En({
    mounted: a,
    exitDuration: n,
    duration: e,
    timingFunction: i,
    onExit: s,
    onEntered: l,
    onEnter: d,
    onExited: u
  });
  return f === 0 ? a ? /* @__PURE__ */ g(Vt, { children: o({}) }) : t ? o({ display: "none" }) : null : p === "exited" ? t ? o({ display: "none" }) : null : /* @__PURE__ */ g(Vt, { children: o(
    Mn({
      transition: r,
      duration: f,
      state: p,
      timingFunction: y
    })
  ) });
}
Wr.displayName = "@mantine/core/Transition";
var M = { root: "m_5ae2e3c", barsLoader: "m_7a2bd4cd", bar: "m_870bb79", "bars-loader-animation": "m_5d2b3b9d", dotsLoader: "m_4e3f22d7", dot: "m_870c4af", "loader-dots-animation": "m_aac34a1", ovalLoader: "m_b34414df", "oval-loader-animation": "m_f8e89c4b" };
const Pn = it(({ className: t, ...r }, e) => /* @__PURE__ */ F(C, { component: "span", className: X(M.barsLoader, t), ...r, ref: e, children: [
  /* @__PURE__ */ g("span", { className: M.bar }),
  /* @__PURE__ */ g("span", { className: M.bar }),
  /* @__PURE__ */ g("span", { className: M.bar })
] })), Ln = it(({ className: t, ...r }, e) => /* @__PURE__ */ F(C, { component: "span", className: X(M.dotsLoader, t), ...r, ref: e, children: [
  /* @__PURE__ */ g("span", { className: M.dot }),
  /* @__PURE__ */ g("span", { className: M.dot }),
  /* @__PURE__ */ g("span", { className: M.dot })
] })), An = it(({ className: t, ...r }, e) => /* @__PURE__ */ g(C, { component: "span", className: X(M.ovalLoader, t), ...r, ref: e })), Br = {
  bars: Pn,
  oval: An,
  dots: Ln
}, On = {
  loaders: Br,
  type: "oval"
}, zn = (t, { size: r, color: e }) => ({
  root: {
    "--loader-size": G(r, "loader-size"),
    "--loader-color": e ? Dt(e, t) : void 0
  }
}), $t = Q((t, r) => {
  const e = I("Loader", On, t), {
    size: n,
    color: a,
    type: o,
    vars: i,
    className: s,
    style: l,
    classNames: d,
    styles: u,
    unstyled: f,
    loaders: p,
    variant: y,
    children: m,
    ...b
  } = e, v = E({
    name: "Loader",
    props: e,
    classes: M,
    className: s,
    style: l,
    classNames: d,
    styles: u,
    unstyled: f,
    vars: i,
    varsResolver: zn
  });
  return m ? /* @__PURE__ */ g(C, { ...v("root"), ref: r, ...b, children: m }) : /* @__PURE__ */ g(
    C,
    {
      ...v("root"),
      ref: r,
      component: p[o],
      variant: y,
      size: n,
      ...b
    }
  );
});
$t.defaultLoaders = Br;
$t.classes = M;
$t.displayName = "@mantine/core/Loader";
const [Wn, St] = ie({
  offsetBottom: !1,
  offsetTop: !1,
  describedBy: void 0,
  getStyles: null,
  inputId: void 0,
  labelId: void 0
});
var R = { wrapper: "m_6c018570", input: "m_8fb7ebe7", section: "m_82577fc2", placeholder: "m_88bacfd0", root: "m_46b77525", label: "m_8fdc1311", required: "m_78a94662", error: "m_8f816625", description: "m_fe47ce59" };
const $r = {}, Bn = (t, { size: r }) => ({
  description: {
    "--input-description-size": r === void 0 ? void 0 : `calc(${W(r)} - ${c(2)})`
  }
}), wt = Q((t, r) => {
  const e = I("InputDescription", $r, t), {
    classNames: n,
    className: a,
    style: o,
    styles: i,
    unstyled: s,
    vars: l,
    size: d,
    __staticSelector: u,
    __inheritStyles: f = !0,
    variant: p,
    ...y
  } = I("InputDescription", $r, e), m = St(), b = E({
    name: ["InputWrapper", u],
    props: e,
    classes: R,
    className: a,
    style: o,
    classNames: n,
    styles: i,
    unstyled: s,
    rootSelector: "description",
    vars: l,
    varsResolver: Bn
  }), v = f && (m == null ? void 0 : m.getStyles) || b;
  return /* @__PURE__ */ g(
    C,
    {
      component: "p",
      ref: r,
      variant: p,
      size: d,
      ...v("description", m != null && m.getStyles ? { className: a, style: o } : void 0),
      ...y
    }
  );
});
wt.classes = R;
wt.displayName = "@mantine/core/InputDescription";
const Fn = {}, jn = (t, { size: r }) => ({
  error: {
    "--input-error-size": r === void 0 ? void 0 : `calc(${W(r)} - ${c(2)})`
  }
}), Ct = Q((t, r) => {
  const e = I("InputError", Fn, t), {
    classNames: n,
    className: a,
    style: o,
    styles: i,
    unstyled: s,
    vars: l,
    size: d,
    __staticSelector: u,
    __inheritStyles: f = !0,
    variant: p,
    ...y
  } = e, m = E({
    name: ["InputWrapper", u],
    props: e,
    classes: R,
    className: a,
    style: o,
    classNames: n,
    styles: i,
    unstyled: s,
    rootSelector: "error",
    vars: l,
    varsResolver: jn
  }), b = St(), v = f && (b == null ? void 0 : b.getStyles) || m;
  return /* @__PURE__ */ g(
    C,
    {
      component: "p",
      ref: r,
      variant: p,
      size: d,
      ...v("error", b != null && b.getStyles ? { className: a, style: o } : void 0),
      ...y
    }
  );
});
Ct.classes = R;
Ct.displayName = "@mantine/core/InputError";
const Sr = {
  labelElement: "label"
}, Vn = (t, { size: r }) => ({
  label: {
    "--input-label-size": W(r),
    "--input-asterisk-color": void 0
  }
}), _t = Q((t, r) => {
  const e = I("InputLabel", Sr, t), {
    classNames: n,
    className: a,
    style: o,
    styles: i,
    unstyled: s,
    vars: l,
    labelElement: d,
    size: u,
    required: f,
    htmlFor: p,
    onMouseDown: y,
    children: m,
    __staticSelector: b,
    variant: v,
    mod: w,
    ...$
  } = I("InputLabel", Sr, e), h = E({
    name: ["InputWrapper", b],
    props: e,
    classes: R,
    className: a,
    style: o,
    classNames: n,
    styles: i,
    unstyled: s,
    rootSelector: "label",
    vars: l,
    varsResolver: Vn
  }), S = St(), _ = (S == null ? void 0 : S.getStyles) || h;
  return /* @__PURE__ */ F(
    C,
    {
      ..._("label", S != null && S.getStyles ? { className: a, style: o } : void 0),
      component: d,
      variant: v,
      size: u,
      ref: r,
      htmlFor: d === "label" ? p : void 0,
      mod: [{ required: f }, w],
      onMouseDown: (k) => {
        y == null || y(k), !k.defaultPrevented && k.detail > 1 && k.preventDefault();
      },
      ...$,
      children: [
        m,
        f && /* @__PURE__ */ g("span", { ..._("required"), "aria-hidden": !0, children: " *" })
      ]
    }
  );
});
_t.classes = R;
_t.displayName = "@mantine/core/InputLabel";
const wr = {}, ar = Q((t, r) => {
  const e = I("InputPlaceholder", wr, t), {
    classNames: n,
    className: a,
    style: o,
    styles: i,
    unstyled: s,
    vars: l,
    __staticSelector: d,
    variant: u,
    error: f,
    mod: p,
    ...y
  } = I("InputPlaceholder", wr, e), m = E({
    name: ["InputPlaceholder", d],
    props: e,
    classes: R,
    className: a,
    style: o,
    classNames: n,
    styles: i,
    unstyled: s,
    rootSelector: "placeholder"
  });
  return /* @__PURE__ */ g(
    C,
    {
      ...m("placeholder"),
      mod: [{ error: !!f }, p],
      component: "span",
      variant: u,
      ref: r,
      ...y
    }
  );
});
ar.classes = R;
ar.displayName = "@mantine/core/InputPlaceholder";
function Hn(t, { hasDescription: r, hasError: e }) {
  const n = t.findIndex((l) => l === "input"), a = t.slice(0, n), o = t.slice(n + 1), i = r && a.includes("description") || e && a.includes("error");
  return { offsetBottom: r && o.includes("description") || e && o.includes("error"), offsetTop: i };
}
const Dn = {
  labelElement: "label",
  inputContainer: (t) => t,
  inputWrapperOrder: ["label", "description", "input", "error"]
}, Yn = (t, { size: r }) => ({
  label: {
    "--input-label-size": W(r),
    "--input-asterisk-color": void 0
  },
  error: {
    "--input-error-size": r === void 0 ? void 0 : `calc(${W(r)} - ${c(2)})`
  },
  description: {
    "--input-description-size": r === void 0 ? void 0 : `calc(${W(r)} - ${c(2)})`
  }
}), ir = Q((t, r) => {
  const e = I("InputWrapper", Dn, t), {
    classNames: n,
    className: a,
    style: o,
    styles: i,
    unstyled: s,
    vars: l,
    size: d,
    variant: u,
    __staticSelector: f,
    inputContainer: p,
    inputWrapperOrder: y,
    label: m,
    error: b,
    description: v,
    labelProps: w,
    descriptionProps: $,
    errorProps: h,
    labelElement: S,
    children: _,
    withAsterisk: k,
    id: T,
    required: x,
    __stylesApiProps: V,
    mod: H,
    ...K
  } = e, Z = E({
    name: ["InputWrapper", f],
    props: V || e,
    classes: R,
    className: a,
    style: o,
    classNames: n,
    styles: i,
    unstyled: s,
    vars: l,
    varsResolver: Yn
  }), L = {
    size: d,
    variant: u,
    __staticSelector: f
  }, A = me(T), st = typeof k == "boolean" ? k : x, J = (h == null ? void 0 : h.id) || `${A}-error`, tt = ($ == null ? void 0 : $.id) || `${A}-description`, rt = A, N = !!b && typeof b != "boolean", D = !!v, O = `${N ? J : ""} ${D ? tt : ""}`, ct = O.trim().length > 0 ? O.trim() : void 0, pt = (w == null ? void 0 : w.id) || `${A}-label`, It = m && /* @__PURE__ */ g(
    _t,
    {
      labelElement: S,
      id: pt,
      htmlFor: rt,
      required: st,
      ...L,
      ...w,
      children: m
    },
    "label"
  ), xt = D && /* @__PURE__ */ g(
    wt,
    {
      ...$,
      ...L,
      size: ($ == null ? void 0 : $.size) || L.size,
      id: ($ == null ? void 0 : $.id) || tt,
      children: v
    },
    "description"
  ), Rt = /* @__PURE__ */ g(te, { children: p(_) }, "input"), Tt = N && /* @__PURE__ */ re(
    Ct,
    {
      ...h,
      ...L,
      size: (h == null ? void 0 : h.size) || L.size,
      key: "error",
      id: (h == null ? void 0 : h.id) || J
    },
    b
  ), Mt = y.map((Et) => {
    switch (Et) {
      case "label":
        return It;
      case "input":
        return Rt;
      case "description":
        return xt;
      case "error":
        return Tt;
      default:
        return null;
    }
  });
  return /* @__PURE__ */ g(
    Wn,
    {
      value: {
        getStyles: Z,
        describedBy: ct,
        inputId: rt,
        labelId: pt,
        ...Hn(y, { hasDescription: D, hasError: N })
      },
      children: /* @__PURE__ */ g(
        C,
        {
          ref: r,
          variant: u,
          size: d,
          mod: [{ error: !!b }, H],
          ...Z("root"),
          ...K,
          children: Mt
        }
      )
    }
  );
});
ir.classes = R;
ir.displayName = "@mantine/core/InputWrapper";
const Gn = {
  variant: "default",
  leftSectionPointerEvents: "none",
  rightSectionPointerEvents: "none",
  withAria: !0,
  withErrorStyles: !0
}, Xn = (t, r, e) => ({
  wrapper: {
    "--input-margin-top": e.offsetTop ? "calc(var(--mantine-spacing-xs) / 2)" : void 0,
    "--input-margin-bottom": e.offsetBottom ? "calc(var(--mantine-spacing-xs) / 2)" : void 0,
    "--input-height": G(r.size, "input-height"),
    "--input-fz": W(r.size),
    "--input-radius": r.radius === void 0 ? void 0 : Qt(r.radius),
    "--input-left-section-width": r.leftSectionWidth !== void 0 ? c(r.leftSectionWidth) : void 0,
    "--input-right-section-width": r.rightSectionWidth !== void 0 ? c(r.rightSectionWidth) : void 0,
    "--input-padding-y": r.multiline ? G(r.size, "input-padding-y") : void 0,
    "--input-left-section-pointer-events": r.leftSectionPointerEvents,
    "--input-right-section-pointer-events": r.rightSectionPointerEvents
  }
}), j = vt((t, r) => {
  const e = I("Input", Gn, t), {
    classNames: n,
    className: a,
    style: o,
    styles: i,
    unstyled: s,
    required: l,
    __staticSelector: d,
    __stylesApiProps: u,
    size: f,
    wrapperProps: p,
    error: y,
    disabled: m,
    leftSection: b,
    leftSectionProps: v,
    leftSectionWidth: w,
    rightSection: $,
    rightSectionProps: h,
    rightSectionWidth: S,
    rightSectionPointerEvents: _,
    leftSectionPointerEvents: k,
    variant: T,
    vars: x,
    pointer: V,
    multiline: H,
    radius: K,
    id: Z,
    withAria: L,
    withErrorStyles: A,
    mod: st,
    ...J
  } = e, { styleProps: tt, rest: rt } = Er(J), N = St(), D = { offsetBottom: N == null ? void 0 : N.offsetBottom, offsetTop: N == null ? void 0 : N.offsetTop }, O = E({
    name: ["Input", d],
    props: u || e,
    classes: R,
    className: a,
    style: o,
    classNames: n,
    styles: i,
    unstyled: s,
    stylesCtx: D,
    rootSelector: "wrapper",
    vars: x,
    varsResolver: Xn
  }), ct = L ? {
    required: l,
    disabled: m,
    "aria-invalid": !!y,
    "aria-describedby": N == null ? void 0 : N.describedBy,
    id: (N == null ? void 0 : N.inputId) || Z
  } : {};
  return /* @__PURE__ */ F(
    C,
    {
      ...O("wrapper"),
      ...tt,
      ...p,
      mod: [
        {
          error: !!y && A,
          pointer: V,
          disabled: m,
          multiline: H,
          "data-with-right-section": !!$,
          "data-with-left-section": !!b
        },
        st
      ],
      variant: T,
      size: f,
      children: [
        b && /* @__PURE__ */ g(
          "div",
          {
            ...v,
            "data-position": "left",
            ...O("section", {
              className: v == null ? void 0 : v.className,
              style: v == null ? void 0 : v.style
            }),
            children: b
          }
        ),
        /* @__PURE__ */ g(
          C,
          {
            component: "input",
            ...rt,
            ...ct,
            ref: r,
            required: l,
            mod: { disabled: m, error: !!y && A },
            variant: T,
            ...O("input")
          }
        ),
        $ && /* @__PURE__ */ g(
          "div",
          {
            ...h,
            "data-position": "right",
            ...O("section", {
              className: h == null ? void 0 : h.className,
              style: h == null ? void 0 : h.style
            }),
            children: $
          }
        )
      ]
    }
  );
});
j.classes = R;
j.Wrapper = ir;
j.Label = _t;
j.Error = Ct;
j.Description = wt;
j.Placeholder = ar;
j.displayName = "@mantine/core/Input";
var Nt = { root: "m_77c9d27d", inner: "m_80f1301b", label: "m_811560b9", section: "m_a74036a", loader: "m_a25b86ee", group: "m_80d6d844" };
const Cr = {
  orientation: "horizontal"
}, Un = (t, { borderWidth: r }) => ({
  group: { "--button-border-width": c(r) }
}), sr = Q((t, r) => {
  const e = I("ButtonGroup", Cr, t), {
    className: n,
    style: a,
    classNames: o,
    styles: i,
    unstyled: s,
    orientation: l,
    vars: d,
    borderWidth: u,
    variant: f,
    mod: p,
    ...y
  } = I("ButtonGroup", Cr, t), m = E({
    name: "ButtonGroup",
    props: e,
    classes: Nt,
    className: n,
    style: a,
    classNames: o,
    styles: i,
    unstyled: s,
    vars: d,
    varsResolver: Un,
    rootSelector: "group"
  });
  return /* @__PURE__ */ g(
    C,
    {
      ...m("group"),
      ref: r,
      variant: f,
      mod: [{ "data-orientation": l }, p],
      role: "group",
      ...y
    }
  );
});
sr.classes = Nt;
sr.displayName = "@mantine/core/ButtonGroup";
const qn = {
  in: { opacity: 1, transform: `translate(-50%, calc(-50% + ${c(1)}))` },
  out: { opacity: 0, transform: "translate(-50%, -200%)" },
  common: { transformOrigin: "center" },
  transitionProperty: "transform, opacity"
}, Qn = {}, Kn = (t, { radius: r, color: e, gradient: n, variant: a, size: o, justify: i, autoContrast: s }) => {
  const l = t.variantColorResolver({
    color: e || t.primaryColor,
    theme: t,
    gradient: n,
    variant: a || "filled",
    autoContrast: s
  });
  return {
    root: {
      "--button-justify": i,
      "--button-height": G(o, "button-height"),
      "--button-padding-x": G(o, "button-padding-x"),
      "--button-fz": o != null && o.includes("compact") ? W(o.replace("compact-", "")) : W(o),
      "--button-radius": r === void 0 ? void 0 : Qt(r),
      "--button-bg": e || a ? l.background : void 0,
      "--button-hover": e || a ? l.hover : void 0,
      "--button-color": l.color,
      "--button-bd": e || a ? l.border : void 0,
      "--button-hover-color": e || a ? l.hoverColor : void 0
    }
  };
}, kt = vt((t, r) => {
  const e = I("Button", Qn, t), {
    style: n,
    vars: a,
    className: o,
    color: i,
    disabled: s,
    children: l,
    leftSection: d,
    rightSection: u,
    fullWidth: f,
    variant: p,
    radius: y,
    loading: m,
    loaderProps: b,
    gradient: v,
    classNames: w,
    styles: $,
    unstyled: h,
    "data-disabled": S,
    autoContrast: _,
    mod: k,
    ...T
  } = e, x = E({
    name: "Button",
    props: e,
    classes: Nt,
    className: o,
    style: n,
    classNames: w,
    styles: $,
    unstyled: h,
    vars: a,
    varsResolver: Kn
  }), V = !!d, H = !!u;
  return /* @__PURE__ */ F(
    or,
    {
      ref: r,
      ...x("root", { active: !s && !m && !S }),
      unstyled: h,
      variant: p,
      disabled: s || m,
      mod: [
        {
          disabled: s || S,
          loading: m,
          block: f,
          "with-left-section": V,
          "with-right-section": H
        },
        k
      ],
      ...T,
      children: [
        /* @__PURE__ */ g(Wr, { mounted: !!m, transition: qn, duration: 150, children: (K) => /* @__PURE__ */ g(C, { component: "span", ...x("loader", { style: K }), "aria-hidden": !0, children: /* @__PURE__ */ g(
          $t,
          {
            color: "var(--button-color)",
            size: "calc(var(--button-height) / 1.8)",
            ...b
          }
        ) }) }),
        /* @__PURE__ */ F("span", { ...x("inner"), children: [
          d && /* @__PURE__ */ g(C, { component: "span", ...x("section"), mod: { position: "left" }, children: d }),
          /* @__PURE__ */ g(C, { component: "span", mod: { loading: m }, ...x("label"), children: l }),
          u && /* @__PURE__ */ g(C, { component: "span", ...x("section"), mod: { position: "right" }, children: u })
        ] })
      ]
    }
  );
});
kt.classes = Nt;
kt.displayName = "@mantine/core/Button";
kt.Group = sr;
var Fr = { root: "m_9e117634" };
const Zn = {}, Jn = (t, { radius: r, fit: e }) => ({
  root: {
    "--image-radius": r === void 0 ? void 0 : Qt(r),
    "--image-object-fit": e
  }
}), cr = vt((t, r) => {
  const e = I("Image", Zn, t), {
    classNames: n,
    className: a,
    style: o,
    styles: i,
    unstyled: s,
    vars: l,
    onError: d,
    src: u,
    radius: f,
    fit: p,
    fallbackSrc: y,
    mod: m,
    ...b
  } = e, [v, w] = Y(!u);
  B(() => w(!u), [u]);
  const $ = E({
    name: "Image",
    classes: Fr,
    props: e,
    className: a,
    style: o,
    classNames: n,
    styles: i,
    unstyled: s,
    vars: l,
    varsResolver: Jn
  });
  return v && y ? /* @__PURE__ */ g(
    C,
    {
      component: "img",
      ref: r,
      src: y,
      ...$("root"),
      onError: d,
      mod: ["fallback", m],
      ...b
    }
  ) : /* @__PURE__ */ g(
    C,
    {
      component: "img",
      ref: r,
      ...$("root"),
      src: u,
      onError: (h) => {
        d == null || d(h), w(!0);
      },
      mod: m,
      ...b
    }
  );
});
cr.classes = Fr;
cr.displayName = "@mantine/core/Image";
const ao = (t) => /* @__PURE__ */ g(j, { ...t, children: t.children });
function io(t) {
  return /* @__PURE__ */ g(kt, { children: t.children });
}
function so(t) {
  return /* @__PURE__ */ g(cr, { ...t });
}
const co = (t) => /* @__PURE__ */ g(Mr, { children: t.children });
function lo() {
  const [t, r] = Y(!1), e = at(null), n = bt(() => r(!0), []), a = bt(() => r(!1), []);
  return B(() => {
    if (e.current)
      return e.current.addEventListener("mouseenter", n), e.current.addEventListener("mouseleave", a), () => {
        var o, i;
        (o = e.current) == null || o.removeEventListener("mouseenter", n), (i = e.current) == null || i.removeEventListener("mouseenter", n);
      };
  }, []), { ref: e, isHover: t };
}
export {
  io as Button,
  oo as H1,
  so as Image,
  ao as Input,
  co as Template,
  lo as useHovered
};
